### PR TITLE
DevOps - Add gitpod as development environment

### PR DESCRIPTION
**Pull request by noklam**
## Description
Why was this PR created?
Gitpod is quite useful as a cloud development environment, it also come in handy for PR review where you have a sandbox created automatically.

## Development notes
What have you changed, and how has this been tested?
- add `test_requirements.txt` since `cookiecutter` is missing
- Add`.gitpod.yml` to setup the develop environment.

## Checklist

- [ ] Read the [contributing](../CONTRIBUTING.md) guidelines
- [ ] Open this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Update the documentation to reflect the code changes
- [ ] Add a description of this change and add your name to the list of supporting contributions in the [`CHANGELOG.md`](../CHANGELOG.md) file. Please respect [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines.
- [ ] Add tests to cover your changes

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

**Comment by noklam** [comment]
> 2 workflows awaiting approval

Is is possible to auto-approve the CI to run?

File: docs/Makefile [hunk 1]
```diff
@@ -18,3 +18,7 @@ help:
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 %: Makefile
 	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+
+test:
+	pytest
```
Comment (Galileo-Galilei) [inline docs/Makefile:23]: What's the expected benefit of this ?

Comment (noklam) [reply docs/Makefile:23]: See https://github.com/Galileo-Galilei/kedro-pandera/pull/9/files#r1279762561, remove it for now.

Comment (Galileo-Galilei) [reply docs/Makefile:23]: I merged it literally 1 mn before this message 😅 Not very important, we can clean it up later

**Pull request merged**